File: pset6/readability/readability.py
import cs50
import math

# make helper functions
# plug values into formula


def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if (text[i].isalpha()):
            letters += 1 
    return float(letters)     


def count_words(text):
    words = 0
    for i in range(len(text)):
        if (text[i].isspace()):
            words += 1
    return float(words) + 1
  
    
def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if (text[i] == '!' or text[i] == '?' or text[i] == '.'):
            sentences += 1
    return float(sentences)
    

# main part of application

text = cs50.get_string("Text: ")
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

l = (letters / words) * 100
s = (sentences / words) * 100  
pre_index = round(0.0588 * l - 0.296 * s - 15.8)
index = math.trunc(pre_index)

if (index < 1):
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

// pseudocode
// need # of letters, words, setences in text
// letters uppercase & lowercase ignore spaces/punctuation
// for loop for length of string (funcion in lecture)
// if upper or lowercase letter increment letter counter (functions to determine type of character)
// words - first nonspace character = word, hit space, look out for next alphabetic character increment word count and repeat
// sentences - any . ! ? indicates a sentence, any time you hit one of these characters increment sentence count
// plug values into formula, round score using round function


int main(void)
{

    string text = get_string("Text: ");
    //totals in text
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    // per 100 words
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    int index = trunc(round(0.0588 * L - 0.296 * S - 15.8));

    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    } 
    else if (index >= 16) 
    {
        printf("Grade 16+\n");
    } 
    else 
    {
        printf("Grade %i\n", index);
    }
}



// helper functions
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return (float) letters;
}

int count_words(string text)
{
    int words = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return (float) words + 1;
}

int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }
    }
    return (float) sentences;
}